### DISCUSSION TITLE
How to show customized 500 Page

### DISCUSSION THREAD

**Discussion by ytliuSVN** [vercel/next.js#23138] [Help] [answered]
I would like to display my **[customized 500 Page](https://nextjs.org/docs/advanced-features/custom-error-page#customizing-the-500-page)** when I got `GET http://localhost:3000/ 500 (Internal Server Error)` in my console. Even I create a `pages/500.js` file still cannot show 500 Page, while `pages/404.js` file can show normally. Should I make some extra configuration? I am using next `^10.0.7`

![Screenshot from 2021-03-17 16-39-59](https://user-images.githubusercontent.com/13102859/111438698-8476ce00-873f-11eb-9b42-486e8718fcaa.png)

Here is my code snippet:

```
export async function getStaticProps() {
  const data = await pricingService.getByCurrency('EUR');

  if (!data) {
    return {
      notFound: true,
    };
  }

  return {
    props: {
      data,
    },
  };
}
```

**Comment by akellbl4** [comment, marked as answer]
Hi!
First of all page with 500 error appears only in production build. You can work on its appearance in dev if open `http://localhost:3000/500`.
You won't see error 500 on pages with `getStaticProps` in production because they are statically optimized and served as HTML.
It means if you have any errors in  `getStaticProps` they will appear only while build time or in background while incremental revalidation.

**Reply by ytliuSVN** [reply]
Got it. Thanks a lot!
